BedrockCore.cpp

```diff
@@ -73,45 +73,53 @@ void BedrockCore::prePeekCommand(unique_ptr<BedrockCommand>& command) {
     STable& content = command->jsonContent;
 
     try {
-        SDEBUG("prePeeking at '" << request.methodLine << "' with priority: " << command->priority);
-        uint64_t timeout = _getRemainingTime(command, false);
-        command->prePeekCount++;
-
-        _db.startTiming(timeout);
+        try {
+            SDEBUG("prePeeking at '" << request.methodLine << "' with priority: " << command->priority);
+            uint64_t timeout = _getRemainingTime(command, false);
+            command->prePeekCount++;
 
-        if (!_db.beginTransaction(SQLite::TRANSACTION_TYPE::SHARED)) {
-            STHROW("501 Failed to begin shared prePeek transaction");
-        }
+            _db.startTiming(timeout);
 
-        // Make sure no writes happen while in prePeek command
-        _db.setQueryOnly(true);
+            if (!_db.beginTransaction(SQLite::TRANSACTION_TYPE::SHARED)) {
+                STHROW("501 Failed to begin shared prePeek transaction");
+            }
 
-        // prePeek.
-        command->reset(BedrockCommand::STAGE::PREPEEK);
-        command->prePeek(_db);
-        SDEBUG("Plugin '" << command->getName() << "' prePeeked command '" << request.methodLine << "'");
+            // Make sure no writes happen while in prePeek command
+            _db.setQueryOnly(true);
 
-        if (!content.empty()) {
-            // Make sure we're not overwriting anything different.
-            string newContent = SComposeJSONObject(content);
-            if (response.content != newContent) {
-                if (!response.content.empty()) {
-                    SWARN("Replacing existing response content in " << request.methodLine);
+            // prePeek.
+            command->reset(BedrockCommand::STAGE::PREPEEK);
+            command->prePeek(_db);
+            SDEBUG("Plugin '" << command->getName() << "' prePeeked command '" << request.methodLine << "'");
+
+            if (!content.empty()) {
+                // Make sure we're not overwriting anything different.
+                string newContent = SComposeJSONObject(content);
+                if (response.content != newContent) {
+                    if (!response.content.empty()) {
+                        SWARN("Replacing existing response content in " << request.methodLine);
+                    }
+                    response.content = newContent;
                 }
-                response.content = newContent;
             }
+        } catch (const SQLite::timeout_error& e) {
+            // Some plugins want to alert timeout errors themselves, and make them silent on bedrock.
+            if (!command->shouldSuppressTimeoutWarnings()) {
+                SALERT("Command " << command->request.methodLine << " timed out after " << e.time() / 1000 << "ms.");
+            }
+            STHROW("555 Timeout prePeeking command");
+        } catch (const SException& e) {
+            _handleCommandException(command, e);
+            command->complete = true;
```

Comment:
This block doesn't make any sense to me. Why are we doing this differently than how peek has done it?

Peek has an inner block that catches `timeout_error` and throws `SException`. Then it has an outer block that catches `SException` and `...` (And also `NotLeading` though that's not relevant here).

Here we've got an inner block that catches `timeout_error`, `SException` and `...`, and in the case of `timeout_error` throws `SException` to an outer block that has the exact same handling as the inner block?

Why don't we just have the `SException` handling in the outer block, like we do for `peek`? And further, why don't we also have `...` in the out block to match peek for consistency?